src/drivers/intel/cavs/hda-dma.c

```diff
@@ -266,36 +289,31 @@ static int hda_dma_copy_ch(struct dma *dma, struct hda_chan_data *chan,
 				DGCS, DGCS_BOR, DGCS_BOR);
 
 	/* make sure that previous transfer is complete */
-	if (chan->direction == DMA_DIR_MEM_TO_DEV) {
-		while (hda_dma_get_free_size(dma, chan->index) <
-		       bytes)
-			idelay(PLATFORM_DEFAULT_DELAY);
-	}
+	while (hda_dma_get_free_size(dma, chan->index) < bytes)
```

Comment:
what happens here if the channel never drains ? should we have a number of tries and then give up (and emit error) ? I'm thinking here that this will allow the FW to continue being responsive and debuggable.

Reply:
We should also get ack from @mmaka1 before merging too.

Reply:
Even if HDA clocks are gated (or is this not possible), sorry just trying to rule out any cases where we may spin forever.

Reply:
Given the fact that this 'while' loop is ran in the context of an interrupt from the sys work queue (timer), and knowing the priorities of the interrupts I don't think it's a scenario that can occur.

Reply:
Actually, you're right. The DMA clock can be gated by the driver. I will add a timeout but in a separate commit if that's okay.

Reply:
ok, separate PR is good for timeout. Is @mmaka1 good for his Ack here ?